server/game/core/Player.js

```diff
@@ -992,6 +992,12 @@ class Player extends GameObject {
      * @param {Object} options
      */
     moveCard(card, targetLocation, options = {}) {
+        // If the card is a resource and it is ready, try to ready another resource instead
+        // and exhaust this one. This should be the desired behavior for most cases.
+        if (card.location === Location.Resource && 'exhausted' in card && !card.exhausted) {
```

Comment:
Just a heads-up I tried to set up type predicates as much as possible to avoid ever needing the "property in object" syntax, especially for cards. In this case we can use `card.canBeExhausted() && !card.exhausted`

Reply:
Nice! I didn't realize it was an option, way nicer this way.